test/functional/p2p_initial_headers_sync.py

```diff
@@ -30,19 +30,9 @@ def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 1
 
-    def setup_chain(self):
-        # This test operates under the assumption that the adjusted time is well ahead of block
-        # time.
-        #
-        # By default when we setup a new chain, we also adjust the mocktime (this is not done in
-        # Bitcoin's test suite), which violates this test's assumption and causes it to fail. We
-        # remedy this by ensuring the test's assumptions are met (i.e. we don't adjust mocktime)
-        #
-        self.log.info("Initializing test directory " + self.options.tmpdir)
-        if self.setup_clean_chain:
-            self._initialize_chain_clean()
-        else:
-            self._initialize_chain()
+    def setup_network(self):
```

Comment:
Using `setup_network()` seems as an ugly solution. Let's add [this refactoring](https://github.com/knst/dash/commit/a7a636c08df06a884f0ee74d93c0c7476a9f581f) better.

If you think that's out-of-scope PR I will create a new PR

Reply:
I like that! Pls submit a PR.

Reply:
it's conflicting with this PR, so, let's @PastaPastaPasta review and merge this one

Reply:
it's here https://github.com/dashpay/dash/pull/6242

Reply:
Let's see if CI is happy there and if it is I'm just going to close this one in fav of yours.